### PR TITLE
Alt Tree View

### PR DESCRIPTION
**Pull request by zahnno**
PR includes:

- Alt View which allows multiple view to be appended to any dao view property.
- Tree Alt View specifically used for the menuDAO to showcase the tableview as well as the tree view.

### CODE REVIEW

**Comment by blakegreen32** [comment]
Looks good to me, great work!

File: src/foam/u2/view/AltView.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    function initE() {
+      this.SUPER();  
+      var self = this;
+      this.altStack = null;      
+      this.altStack = this.Stack.create();
+      this.viewChoices$.sub(this.changeView);
+
+      this.views.forEach(function(view){
+        view[0].data = view[0].data ? view[0].data : self.data;
+      })
```
Comment (kgrgreer) [inline src/foam/u2/view/AltView.js]: Add ;

File: src/foam/u2/view/AltView.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+      this.addClass(this.myClass())
+      this.startContext({data: this})
+        this.start()
+          .add(this.VIEW_CHOICES)
+        .end()
+      .endContext()
+      .start('div').addClass('stack-wrapper')
+        .tag({class: 'foam.u2.stack.StackView', data: this.altStack, showActions: false})
+      .end()
```
Comment (kgrgreer) [inline src/foam/u2/view/AltView.js]: Add ;
Remove blank line.

File: src/foam/u2/view/TreeAltView.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+*/
+
+foam.CLASS({
+  package: 'foam.u2.view',
```
Comment (kgrgreer) [inline src/foam/u2/view/TreeAltView.js]: Move to foam.nanos.menu package.

File: src/foam/u2/view/AltView.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+      name: 'viewChoices',
+      view: function(_, X) {
+        return foam.u2.view.ChoiceView.create({
+          choices: X.data.views
+        });
+      }
+    },
+  ],
+
+  axioms: [
```
Comment (kgrgreer) [inline src/foam/u2/view/AltView.js]: Use new css: style for CSS.

File: src/services [hunk 5]
```diff
@@ -30,7 +30,7 @@ p({"class":"foam.nanos.boot.NSpec", "name":"regionDAO",                   "lazy"
...
...nu.DAOMenu\", \"daoKey\":\"nSpecDAO\" }},{  \"parent\":\"admin\", \"id\":\"export\", \"label\":\"Export Drivers\", \"handler\":{  \"class\":\"foam.nanos.menu.DAOMenu\", \"daoKey\":\"exportDriverRegistryDAO\" }},{  \"parent\":\"admin\", \"id\":\"menus\", \"label\":\"Menus\", \"handler\":{  \"class\":\"foam.nanos.menu.DAOMenu\", \"daoKey\":\"menuDAO\", \"summaryView\":{ \"class\": \"foam.nanos.menu.TreeAltView\"} }},{  \"parent\":\"admin\", \"id\":\"scripts\", \"label\":\"Scripts\", \"handler\":{  \"class\":\"foam.nanos.menu.DAOMenu\", \"daoKey\":\"scriptDAO\" }},{  \"parent\":\"admin\", \"id\":\"tests\", \"label\":\"Tests\", \"handler\":{  \"class\":\"foam.nanos.menu.DAOMenu\", \"daoKey\":\"testDAO\", \"summaryView\":{ \"class\":\"foam.nanos.test.TestBorder\" } }},{  \"parent\":\"admin\", \"id\":\"cron\", \"label\":\"Cron Jobs\", \"handler\":{  \"class\":\"foam.nanos.menu.DAOMenu\", \"daoKey\":\"cronDAO\" }},{  \"parent\":\"admin\", \"id\":\"p\", \"label\":\"Performance\", \"handler\":{  \"class\":\"foam.nanos.menu.DAOMenu\", \"daoKey\":\"pmInfoDAO\", \"summaryView\":{ \"class\":\"foam.nanos.pm.PMTableView\" } }},{  \"parent\":\"admin\", \"id\":\"log\", \"label\":\"View Logs\"}]}"})
```
Comment (kgrgreer) [inline src/services]: Needs to be updated. Menus are now stored in menus.

Comment (kgrgreer) [inline src/services]: Delete/revert this line change.

**Pull request merged**